### PR TITLE
[Android] Make XWalkSettings public

### PR DESCRIPTION
**Pull request by lincsoon**
Expose XWalkSettings to embedding API

BUG=XWALK-4395

### CODE REVIEW

**Comment by lincsoon** [comment]
@axinging @fujunwei @xzhan96 @XiaosongWei @wuhengzhi

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkSettingsInternal.java [hunk 1]
```diff
@@ -144,7 +145,15 @@ private void updateWebkitPreferencesLocked() {
...
 
-    public XWalkSettings(Context context, WebContents webContents,
+    // Never use this constructor.
+    // It is only used in XWalkSettingsBridge.
+    XWalkSettingsInternal() {
+        mContext = null;
+        mEventHandler = null;
+    }
+
+
```
Comment (axinging) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkSettingsInternal.java]: One extra empty line.

**Comment by axinging** [comment]
Except the above nit, LGTM.
One comments,  we should file a task to update the online API documents.

**Comment by wuhengzhi** [comment]
Some test cases which used ```XWalkViewInternal.getSettings()``` should be moved to ```xwalk/core``` since there is no test case called ```XWalkView.getSettings()```. Or add test case to check ```XWalkView.getSettings()```.

**Comment by xzhan96** [comment]
is the purpose of keeping the two ways (public XWalkSettings & API in XWalkView ) to get settings is for compatibility?

**Comment by lincsoon** [comment]
Added test case

**Comment by lincsoon** [comment]
@xzhan96 We will implement new features only in XWalkSettings in the future. But duplicated methods in XWalkView will remain

**Comment by wuhengzhi** [comment]
LGTM for other parts, thanks.

**Pull request merged**